pkgs/applications/audio/go-musicfox/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+, clangStdenv
+, pkg-config
+, alsa-lib
+, flac
+}:
+
+buildGoModule.override { stdenv = clangStdenv; } rec {
```

Comment:
Any reason to use clangStdenv here?

Reply:
I don't know why, but it can not build in `stdenv`.

https://github.com/anhoder/go-musicfox/blob/master/.goreleaser.yaml#L12

Reply:
They are also using gcc: https://github.com/anhoder/go-musicfox/blob/7002706f6a0da3a8b372fdfac3e716b5c282ebce/.goreleaser.yaml#L14. Can you provide logs or other information?

Reply:
```
› nix log /nix/store/xs7a0m2clzbvrmfxf2xsqk9wq6cka78v-go-musicfox-3.6.1.drv
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/g38dbphhq31jlvqhqhbd39bs336pbrfb-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage ./cmd/musicfox.go
# github.com/cocoonlife/goflac
gcc: error: language objective-c not recognized
gcc: error: language objective-c not recognized
```

Reply:
That's it: https://github.com/NixOS/nixpkgs/blob/2bf344a7a2e27912051e6b1c419f1712b6e60f3f/pkgs/development/compilers/gcc/12/default.nix#L4
Our gcc only has objc support on darwin.

Reply:
Please add a comment about this situation above the override. Let's see if we can add objc support by default to gcc later. cc @trofi as you seem to be familiar with that sort of things.

Reply:
Heh, yeah. That's surprising that we do have `langObjC`, but don't have it used anywhere in the tree. The following hack makes the `go-musicfox` compile for me. Don't know if it actually runs.

```diff
--- a/pkgs/applications/audio/go-musicfox/default.nix
+++ b/pkgs/applications/audio/go-musicfox/default.nix
@@ -5,10 +5,11 @@
 , pkg-config
 , alsa-lib
 , flac
+, gobjc
 }:

 # gcc only supports objc on darwin
-buildGoModule.override { stdenv = clangStdenv; } rec {
+buildGoModule rec {
   pname = "go-musicfox";
   version = "3.6.1";

@@ -33,6 +34,7 @@ buildGoModule.override { stdenv = clangStdenv; } rec {

   nativeBuildInputs = [
     pkg-config
+    gobjc
   ];

   buildInputs = [
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 5bf8b93ac66..0ceb6502f9f 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -14546,6 +14546,14 @@ with pkgs;
     profiledCompiler = false;
   });

+  gobjc = wrapCC (gcc12.cc.override {
+    name = "gobjc";
+    langObjC = true;
+    langCC = false;
+    langC = false;
+    profiledCompiler = false;
+  });
+
   libgccjit = gcc.cc.override {
     name = "libgccjit";
     langFortran = false;
```